src/unix/fs.c

```diff
@@ -1439,7 +1439,8 @@ static int uv__fs_statx(int fd,
   buf->st_nlink = statxbuf.stx_nlink;
   buf->st_uid = statxbuf.stx_uid;
   buf->st_gid = statxbuf.stx_gid;
-  buf->st_rdev = statxbuf.stx_rdev_major;
+  buf->st_rdev = ((statxbuf.stx_rdev_minor & 0xFF) | (statxbuf.stx_rdev_major << 8)
+                 | ((statxbuf.stx_rdev_minor & ~0xFF) << 12));
```

Comment:
Suggestion: `#include <sys/sysmacros.h>` at the top (guarded on `#ifdef __linux__`) and then do this:
```suggestion
  buf->st_rdev = makedev(statxbuf.stx_rdev_major, statxbuf.stx_rdev_minor);
```
It probably makes sense to update `buf->st_dev` in that fashion as well.